docs/architecture/adr-008.md

```diff
@@ -0,0 +1,191 @@
+1. Unpack and validate bridge parameters from the transaction or event;
+2. If the token source is Cronos EVM, invoke `keeper.BurnVouchersToEvmCoins()`; Otherwise, invoke `keeper.ConvertVouchersToEvmCoins()`^2;
+3. Emit Cosmos events as needed.
+
+^2 BurnVouchersToEvmCoins will burn the native token, mint the local EVM tokens and send the token to the user EVM address. More details of the new keeper methods will be explained below.
+
+#### D. Native Denom Naming
+
+The denom naming of the native token with Cronos EVM as source is
+```
+evm0x{smartContractAddress}
+```
+
+The chain must avoid naming any native asset with the prefix `evm0x` in the future.
+
+### 4. Keeper Methods Changes
+
+### Method Updates
+
+#### GetDenomByContract
+
+```go
+GetDenomByContract(ctx sdk.Context, contract common.Address)
+    (source int, denom string, found bool)
+```
+
+`source` is added in the returned value to denote the source of the denom to smart contract address mapping
+
+##### GetContractByDenom
+
+```go
+GetContractByDenom(ctx sdk.Context, denom string)
+    (source int, contract common.Address, found bool)
+```
+
+`source` is added in the returned value to denote the source of the denom to smart contract address mapping
+
+##### ConvertCoinFromNativeToCRC20
+
+In the current design, the method permits only the denom is one of the IBC and Gravity Bridge denoms before doing any conversion. In this ADR, we want to loosen this validation check. If such a requirement exists, it should be performed in the IBC hooks.
+
+### New Method
+
+##### BurnVouchersToEvmCoins
+
+```go
+BurnVouchersToEvmCoins(ctx sdk.Context, receiver string, coins sdk.Coins) error
+```
+
+This method works for local EVM tokens. It burns the previously-minted native token from the `receiver` address. This method will then invoke the smart contract to mint the EVM tokens to the EVM address of the `receiver`.
+
+The working mechanism is as follows:
+1. Call `keeper.GetContractByDenom()` to validate and get the token source and smart contract address of the provided native token denom;
+2. Burn the voucher native token from the `receiver` Cosmos address;
+3. Invoke the corresponding smart contract `mint_by_cronos_module` method to mint the EVM token to `receiver` EVM address.
+
+## Status
+
+Proposed
+
+## Consequences
+
+### Positive
+
+- Support local EVM token to native token conversion.
+- No new key value pairs added and storage increment kept at minimum.
+- Compatible with existing CRC20/CRC21 contract standards.
+
+### Negative
+
+- No flexible control of token conversion. For example, we cannot restrict certain token cannot be bridged over particular protocol. It relies on the bridge module to manage the restriction list.
+- Potentially complicated chain upgrade logic.
+
+### Neutral
+
+- Multi-sourced token is not supported. This is a deliberate design decision to avoid complicated supply management logic.
+
+## References
```

Comment:
maybe one reference to look at: https://github.com/evmos/evmos/tree/main/x/erc20/spec